### PR TITLE
npm audit in CI

### LINKED ISSUE #1126: Add `npm audit` to Jenkins pipeline
**Is your feature request related to a problem? Please describe.**
Blocked on removing Razzle (#715) (which has a high vulnerability).

Once that's done, we should hopefully be vulnerability-free. We should continue to do so! This issue helps ensure that.

**Describe the solution you'd like**
Make `npm audit` part of the Jenkins pipeline so that if known vulnerabilities are introduced or discovered, our build fails and we can prioritise an urgent fix.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Should see `npm audit` run in the Jenkins output with similar output to the following:

```
$ npm audit

                       === npm audit security report ===

found 0 vulnerabilities
 in 43607 scanned packages
```

(And if we `npm install razzle` and `npm audit` again, we should see a failure)

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by sareh**
Resolves #1126

- Add `audit-ci` to run in script `audit:ci`, to error out of the build if there is a vulnerability of severity `low` or greater. 
- Uses `npm run audit:ci` in Travis & Jenkins only against PRs.  

I looked at other packages such as [npm-audit-ci](https://www.npmjs.com/package/npm-audit-ci) &  [npm-audit-ci-wrapper](https://www.npmjs.com/package/npm-audit-ci-wrapper), but `audit-ci` had the most users, is actively maintained by a team at IBM, so will likely continue to be maintained in future.  The ability to pass in flags for `--low` `--medium` `--high` or `--critical` is useful for our case. 

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Comment by sareh** [comment]
This build on Travis passed here: https://travis-ci.org/bbc/simorgh/jobs/477454850#L508-L529
and on Jenkins: https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/simorgh/detail/npm-audit-ci/1/pipeline/

Now I will purposefully add a dependency that has a vulnerability, to demonstrate that both Travis & Jenkins will exit with a non-zero exit code and cause the build to fail.

**Comment by sareh** [comment]
I found this advisory stating that `merge@1.2.0` has vulnerabilities of low severity. https://www.npmjs.com/advisories/722  I will add this as a dependency and expect the Travis & Jenkins pipelines to fail.

**Comment by sareh** [comment]
Jenkins job exited and failed correctly. Also correctly reported 9 low-severity vulnerabilities.

screen shot Jenkins CI job
https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/simorgh/detail/npm-audit-ci/2/pipeline

Travis CI job also exited and failed correctly, reporting the 9 low-severity vulnerabilities:
https://travis-ci.org/bbc/simorgh/jobs/477464165
screen shot travis CI job

File: package.json [hunk 1]
```diff
@@ -114,6 +115,7 @@
   "devDependencies": {
     "@storybook/react": "4.0.0-alpha.8",
     "amphtml-validator": "^1.0.23",
+    "audit-ci": "^1.2.1",
```
Comment (ChrisBAshton) [inline package.json:118]: What was the reason for adding a dependency when we can run `npm audit` directly?

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,7 @@ none:
 install:
 	npm --version; node --version;
 	cd ${APP_DIRECTORY}; npm ci;
+	cd ${APP_DIRECTORY}; npm run audit:ci;
```
Comment (dr3) [inline Makefile]: This doesn't feel like an install step, but more of a tests step

**Review by dr3 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sareh) [reply package.json:118]: I looked at `npm audit` directly initially.  It just reports and does not ever give a non-zero response code.

It is possible to use `npm audit --parseable` to give an output, however it doesn't show you the number of vulnerabilities you get - there should be 9 low-severity vulnerabilities here: 
`npm audit --parseable` output:
```
install	merge	low	npm install merge@1.2.1	Prototype pollution	https://nodesecurity.io/advisories/722	merge	N
update	merge	low	npm update merge --depth 9	Prototype pollution	https://nodesecurity.io/advisories/722	jest>jest-cli>jest-haste-map>sane>exec-sh>merge	N
```
An alternative `npm audit --json` gives more verbose output, including the number and severity of vulnerabilities. 
To use this, I'd need to parse the JSON output. It appears that `audit-ci` already handles this and has test coverage too: https://github.com/IBM/audit-ci/blob/master/lib/npm-auditer.js 

So I decided to use this here.

**Review by dr3 — DISMISSED** [review]

**Review by bcmn — DISMISSED** [review]
Looks really good! Thanks :)

**Review by dr3 — APPROVED** [review]

**Review by bcmn — APPROVED** [review]

**Pull request merged**